File: examples/org.specapi.plugins.dotnet.example/dotnet-gen/Org.Specapi.Github/ListUserRepositoriesRequest.cs
using System.Collections.Specialized;
using System.Linq;
using System;
using System.Net;
namespace Org.Specapi.Github
{
    public partial class ListUserRepositoriesRequest
    {
        public string User { get; set; }
        bool _typeParamSet = false;
        string _typeParam;
        public string Type 
        { 
            get 
            {
                return _typeParam;
            } 
            set 
            {
                _typeParamSet = true;
                _typeParam = value;
            }
        }
        
        bool _sortParamSet = false;
        string _sortParam;
        public string Sort 
        { 
            get 
            {
                return _sortParam;
            } 
            set 
            {
                _sortParamSet = true;
                _sortParam = value;
            }
        }
        
        bool _directionParamSet = false;
        string _directionParam;
        public string Direction 
        { 
            get 
            {
                return _directionParam;
            } 
            set 
            {
                _directionParamSet = true;
                _directionParam = value;
            }
        }
        
        
        public ListUserRepositoriesRequest(string user) {
            User = user;
        }
        
        public Uri CreateUri (string baseUrl)
        {
            String path = String.Format ("/users/{0}/repos", User);
            String url = baseUrl + path;
            
            var uriBuilder = new UriBuilder(url);
            NameValueCollection query = new NameValueCollection ();
            if(_typeParamSet) {
                query ["type"] = _typeParam;
            }
            if(_sortParamSet) {
                query ["sort"] = _sortParam;
            }
            if(_directionParamSet) {
                query ["direction"] = _directionParam;
            }
            uriBuilder.Query = string.Join("&", query.AllKeys.Select(key => string.Format("{0}={1}", WebUtility.UrlEncode(key), WebUtility.UrlEncode(query[key]))));
            return uriBuilder.Uri;
        }
    }
}

File: examples/org.specapi.plugins.android.example/src-gen/org/specapi/github/ListUserRepositoriesRequest.java
package org.specapi.github;


import android.net.Uri;
import org.specapi.android.net.ServiceRequest;

public class ListUserRepositoriesRequest extends ServiceRequest {
	
	private static final String PATH = "/users/nullusernull/repos";
	
	private final String userSegment;
	
	private String typeParam;
	private boolean typeParamSet;
	private String sortParam;
	private boolean sortParamSet;
	private String directionParam;
	private boolean directionParamSet;
		
	public ListUserRepositoriesRequest setTypeParam(String value) {
		this.typeParam = value;
		this.typeParamSet = true;
		return this;
	}
	
	public boolean isTypeParamSet() {
		return typeParamSet;
	}
	public ListUserRepositoriesRequest setSortParam(String value) {
		this.sortParam = value;
		this.sortParamSet = true;
		return this;
	}
	
	public boolean isSortParamSet() {
		return sortParamSet;
	}
	public ListUserRepositoriesRequest setDirectionParam(String value) {
		this.directionParam = value;
		this.directionParamSet = true;
		return this;
	}
	
	public boolean isDirectionParamSet() {
		return directionParamSet;
	}
		
	public ListUserRepositoriesRequest(String userSegment){
		this.userSegment = userSegment;
	}
	
	@Override
	public String createUrl(String baseUrl){
		Uri.Builder uriBuilder = Uri.parse(baseUrl + String.format(PATH, userSegment)).buildUpon();
			
		if(typeParamSet){
			uriBuilder.appendQueryParameter("type", typeParam);
		}
		if(sortParamSet){
			uriBuilder.appendQueryParameter("sort", sortParam);
		}
		if(directionParamSet){
			uriBuilder.appendQueryParameter("direction", directionParam);
		}
		
		return uriBuilder.toString();			
	}
}
